packages/shared/src/components/drawers/Drawer.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ReactElement, ReactNode, useEffect, useState } from 'react';
+import classNames from 'classnames';
+import { RootPortal } from '../tooltips/Portal';
+import useDebounce from '../../hooks/useDebounce';
+
+export enum DrawerPosition {
+  Bottom = 'bottom',
+  Top = 'top',
+}
+
+interface ClassName {
+  overlay?: string;
+  drawer?: string;
+}
+
+export interface DrawerProps {
+  children: ReactNode;
+  className?: ClassName;
+  position?: DrawerPosition;
+  closeOnOutsideClick?: boolean;
+  onClose(): void;
+}
+
+const drawerPositionToClassName: Record<DrawerPosition, string> = {
+  [DrawerPosition.Bottom]: 'bottom-0 rounded-t-16',
+  [DrawerPosition.Top]: 'top-0 rounded-b-16',
+};
+
+const animatePositionClassName: Record<DrawerPosition, string> = {
+  [DrawerPosition.Bottom]: 'translate-y-full',
+  [DrawerPosition.Top]: '-translate-y-full',
+};
+
+export function Drawer({
+  children,
+  className = {},
+  position = DrawerPosition.Bottom,
+  closeOnOutsideClick = true,
+  onClose,
+}: DrawerProps): ReactElement {
+  const [hasAnimated, setHasAnimated] = useState(false);
+  const [animate] = useDebounce(() => setHasAnimated(true), 1);
+
+  useEffect(() => {
+    if (!hasAnimated) {
+      animate();
+    }
+  }, [animate, hasAnimated]);
```

Comment:
Why do we need this code here? Seems rather hard to follow.

Reply:
When animating the entrance of conditionally rendered elements, we need to ensure the element is available before running the animation itself. Otherwise, it will not work.

Reply:
Should we use a ref instead? How does this make sure the element is available?

Reply:
On line 42, we wrapped the function in a debounce with 1ms, which ensures the elements are already written in the DOM.

We need the re-render to execute the animation, hence we can only use `useState`.

Reply:
If you use a `ref` on the main drawer element, we don't really need a debounce and it actually ensures that we have the element in the DOM. Or am I missing something? :)

Reply:
Ah, I thought you meant `useRef` over `useState`. That is actually a great idea. Let me check that quickly.

Reply:
We would still need the debounce though that was a better approach since we are now one less `useEffect`. Thanks for the suggestion! 🚀